controllers/storagecluster/cephcluster.go

```diff
@@ -531,11 +531,18 @@ func newStorageClassDeviceSets(sc *ocsv1.StorageCluster, serverVersion *version.
 					if noPreparePlacement {
 						in := getPlacement(sc, "osd-prepare-tsc")
 						(&in).DeepCopyInto(&preparePlacement)
+					}
 
-						if len(topologyKeyValues) >= replica {
-							// If topologyKey is not host, append additional topology spread constarint to the
-							// default preparePlacement. This serves even distribution at the host level
-							// within a failure domain (zone/rack).
+					if len(topologyKeyValues) >= replica {
+						if portable && !strings.Contains(topologyKey, "zone") {
+							placement.TopologySpreadConstraints = append(placement.TopologySpreadConstraints, placement.TopologySpreadConstraints[0])
```

Comment:
This is a little funky. I think it would be a bit easier to read if you create a new variable like `newTSC`, set it to constraint `[0]`, modified it, then end with `[0] = {newTSC, [0]}`. Thoughts?